### ISSUE TITLE
containerengine_clusters_data_source throws errors on setting state as list

### ISSUE THREAD

**Issue by fosrias** [terraform-providers/terraform-provider-oci#600]
### Terraform Version
Terraform v0.11.8

### OCI Provider Version
terraform-provider-oci 2.2.3

### Description:
The following data source definition will crash terraform. 

### Terraform Plan
```
data "oci_containerengine_clusters" "filtered" {
    compartment_id = "${var.compartment_id}"

    #Filter
    name = "${var.name}"
    state = ["ACTIVE"]
}
```
If you make state a string only (not a list) terraform throws an error:

```
Error: data.oci_containerengine_clusters.filtered: state: should be a list
```

Schema shows it should be a list of strings: https://github.com/terraform-providers/terraform-provider-oci/blob/master/provider/containerengine_clusters_data_source.go#L25-L31

Crashlog pertinent section is:
```
2018-09-17T12:14:48.991-0700 [DEBUG] plugin.terraform-provider-oci_v2.2.3: panic: interface conversion: interface {} is []interface {}, not []string
2018-09-17T12:14:48.991-0700 [DEBUG] plugin.terraform-provider-oci_v2.2.3: 
2018-09-17T12:14:48.991-0700 [DEBUG] plugin.terraform-provider-oci_v2.2.3: goroutine 43 [running]:
2018-09-17T12:14:48.991-0700 [DEBUG] plugin.terraform-provider-oci_v2.2.3: github.com/oracle/terraform-provider-oci/provider.(*ClustersDataSourceCrud).Get(0xc4206e43e0, 0x414fed0, 0x0)
2018-09-17T12:14:48.991-0700 [DEBUG] plugin.terraform-provider-oci_v2.2.3:     /projects/go/src/github.com/oracle/terraform-provider-oci/provider/containerengine_clusters_data_source.go:74 +0x8e4
...
```
There is no test for `state` in the provider tests ATM. 

This shows up in provider 2.2.4 as well.

**Comment by alexng-canuck** [comment]
Adding @codycushing to take a look.

**Comment by alexng-canuck** [comment]
Thanks @fosrias for sharing this. This issue is understood now. 

The bug is in this part of the datasource's Get implementation: 
https://github.com/terraform-providers/terraform-provider-oci/blob/c839d893dc4f4c95bb8a75395c03d3395deda706/oci/containerengine_clusters_data_source.go#L74

The underlying type of `states` is an `[]interface{}` filled with strings and it's being converted to a `[]string`. This type of conversion isn't supported in golang. The proper fix is to convert each element of the list to a string individually.

We are working on a fix for this.

**Comment by alexng-canuck** [comment]
Hi @fosrias ,

This issue has been fixed in the latest v3.2.0 release. Can you pick up this version and try it out?

If you need assistance on upgrading from v2.2.3 to the latest release, please refer to this guide: https://www.terraform.io/docs/providers/oci/guides/version-3-upgrade.html

Closing this issue. If you encounter any issues related to this, please reopen the bug.

**Comment by fosrias** [comment]
@alexng-canuck Confirmed working with 3.2.
